### PR TITLE
feat(build): use Docker Build Cloud for image build

### PR DESCRIPTION
**Pull request by gustavovalverde**
## Motivation

We've been using extra large machines in GitHub (like our `ubuntu-latest-xl`) to be able to build our CI Docker images, as the build time is considerably large, but this has also increased our billing use above $1,000. For this reason we've considered moving this specific task to the new Docker Build Cloud service (https://www.docker.com/products/build-cloud/)

#### Check before marking the PR as ready for review:
  - [x] Will the PR name make sense to users?
  - [x] Does the PR have a priority label?
  - [x] Have you added or updated tests?
  - [x] Is the documentation up to date?

##### For significant changes:
  - [ ] Is there a summary in the CHANGELOG?
  - [ ] Can these changes be split into multiple PRs?

_If a checkbox isn't relevant to the PR, mark it as done._

### Specifications

https://docs.docker.com/build/cloud/ci/

## Solution

Configure the `setup-buildx-action` to use a remote `buildx` defined at https://build.docker.com/accounts/zfnd/builders

### Testing

The image should get built, not slower than using our actual implementation

### Reviewer Checklist

Check before approving the PR:
  - [ ] Does the PR scope match the ticket?
  - [ ] Are there enough tests to make sure it works? Do the tests cover the PR motivation?
  - [ ] Are all the PR blockers dealt with?
        PR blockers can be dealt with in new tickets or PRs.

### CODE REVIEW

File: .github/workflows/sub-build-docker-image.yml [hunk 1]
```diff
@@ -171,17 +174,3 @@ jobs:
...
-          #
-          # We use multiple cache sources to confirm a cache hit, starting from a per-branch cache.
-          # If there's no hit, we continue with a `main` branch cache, which helps us avoid
-          # rebuilding cargo-chef, most dependencies, and possibly some Zebra crates.
-          #
-          # The caches are tried in top-down order, the first available cache is used:
-          # https://github.com/moby/moby/pull/26839#issuecomment-277383550
-          cache-from: |
-            type=registry,ref=us-docker.pkg.dev/${{ vars.GCP_PROJECT }}/zebra-caching/${{ inputs.image_name }}${{ inputs.tag_suffix }}:${{ env.GITHUB_REF_SLUG_URL }}-cache
-            type=registry,ref=us-docker.pkg.dev/${{ vars.GCP_PROJECT }}/zebra-caching/${{ inputs.image_name }}${{ inputs.tag_suffix }}:main-cache
```
Comment (arya2) [inline .github/workflows/sub-build-docker-image.yml:185]: Was it doing incremental builds from the main branch, and if it was, is it still doing incremental builds from the main branch?

**Review by arya2 — APPROVED** [review]
Looks good and seems to work, thank you!

File: .github/workflows/sub-build-docker-image.yml (hunk 1 above)
Comment (gustavovalverde) [reply .github/workflows/sub-build-docker-image.yml:185]: This was supposed to work, it technically works, but in reality (with Rust) this issue prevents that from happening:
- https://github.com/ZcashFoundation/zebra/issues/6169

I added an extra comment there so we don't forget it.
https://github.com/ZcashFoundation/zebra/issues/6169#issuecomment-2162814255

**Comment by gustavovalverde** [comment]
@mergifyio refresh

**Pull request merged**